### PR TITLE
[Workload] Refresh workload type "deployment" to make it as a workload in KubeVela

### PR DESCRIPTION
**Pull request by zzxwill**
Extend `deployment` to be a workload type of KubeVela and add example
on how to deploy and access an application with this workload type.

### Checklist

I have:
- [x] Title of the PR starts with OAM type (e.g. `[Workload]` or `[Trait]` or `[Scope]`).
- [x] Updated/Added any relevant [documentation] and [examples].
- [ ] Unit/E2E Tests added.

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]
LGTM， good job

File: workloads/deployment/workloadDefinition.yaml [hunk 1]
```diff
@@ -1,7 +1,89 @@
 apiVersion: core.oam.dev/v1alpha2
 kind: WorkloadDefinition
 metadata:
-  name: deployments.apps
+  name: deployment
```
Comment (wonderflow) [inline workloads/deployment/workloadDefinition.yaml:4]: 把这个完整的放到这里吧 https://github.com/oam-dev/catalog/tree/master/registry

Comment (zzxwill) [reply workloads/deployment/workloadDefinition.yaml:4]: `registry` 目录是用来做什么的？

Comment (wonderflow) [reply workloads/deployment/workloadDefinition.yaml:4]: 就是 capability center 的repo

**Pull request merged**